Discussion (#17)
greenelab/text_mined_hetnet_manuscript: content/05.discussion.md

## Discussion

We tested the feasibility of re-using label functions to extract relationships from literature.
Through our sampling experiment we found that adding relevant label functions increases prediction performance (shown in the on-diagonals of Figures {@fig:generative_model_auroc} and {@fig:generative_model_auprc}).
We found that label functions designed from relatively related edge types can increase performance (seen when GiG label functions predicts CbG and vise versa).
We also found that one edge type (DaG) is agnositic to label function source (Figures {@fig:generative_model_auroc} and {@fig:generative_model_auprc}). 

> An interesting finding is performance increases when adding a single label funciton to the generative model baseline.

Reviewer: funciton is misspelled. Has "generative model baseline" been explained elsewhere? It's not entirely clear here that these label functions are in addition to the knowledge base labeling functions.

Initially we thought that adding a bit of noise aided the model, but this turns out to not be the case (Figure {@fig:random_label_function_auroc} and {@fig:random_label_function_auprc}).
This leads to the lingering question: why does performance drastically increase when adding a single label function to the distant supervision baseline?

The discriminative model didn't work as intended. 
Majority of the time the discriminative model underperformed the generative model (Figures {@fig:discriminative_model_auroc} and {@fig:discriminative_model_auprc}).

> Potential reasons for this is the discriminative model overfits to the generative model's predictions and there is a negative class bias for our datasets (Table {@tbl:candidate-sentences}).

Reviewer: > Potential reasons for this is

is should be are because the subject of "reasons" is plural

The negative bias can skew our results, but overall the discriminative model requires more tuning and regularization to increase performance.

Deep learning models have a tendency to be overconfident in their predictions.
To account for this problem we calibrated our discriminative model.

> Despite our efforts, model calibration didn't help performance (Figure {@fig:discriminative_model_calibration}).

Reviewer: When you say "didn't help performance", how are you judging performance? If using a method that only considers rank (like AUROC or AUPRC), then calibration CANNOT have an effect on performance.

## Discussion

We tested the feasibility of re-using label functions to extract relationships from literature.
Through our sampling experiment we found that adding relevant label functions increases prediction performance.
This can been seen in the on-diagonals of generative model performance plots (Figure {@fig:generative_model_auroc} and {@fig:generative_model_auprc}).

> An interesting finding is performance increases when text patterns and domain heuristic label functions are added on top of the distant supervision label functions. 

Reviewer 2: ```suggestion
```

> This increase happens because curated databases have low coverage and end up missing information.  

Reviewer 2: Do you have evidence for this? If so, I missed it. If not, it's probably best to delete it.

> Some relationships are easier to predict than others.

Reviewer 2: ```suggestion
We also observed that certain relationships were easier to predict than others.
```

> For example, we achieved a significantly high area under the receiver operating curve (AUROC) and area under the precision recall curve (AUPR) for Compound-treats-Disease (CtD) (Figure {@fig:generative_model_auroc} and {@fig:generative_model_auprc}).

Reviewer 2: ```suggestion
For example, we achieved a significantly high AUROC and AUPR for CtD than other edge types (Figures {@fig:generative_model_auroc} and {@fig:generative_model_auprc}).
```

> Reason for high performance is that wording is pretty straight forward when abstracts talk about this particular edge type.

Reviewer 2: Do you have evidence for this? If not, should delete.

> Regarding Disease-associates-Gene (DaG), this edge type has a broad range of keywords that can infer associations.

Reviewer 2: There are a lot of assertions here and I didn't see evidence for them as I read the paper. For these claims where does the evidence come from? If it's not in the paper, it either needs to be in the paper or the claim should be removed.
